compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -1429,6 +1434,42 @@ public BLangNode transform(SyncSendActionNode syncSendActionNode) {
         return workerSendExpr;
     }
 
+    @Override
+    public BLangNode transform(ImplicitAnonymousFunctionExpressionNode implicitAnonymousFunctionExpressionNode) {
+        BLangArrowFunction arrowFunction = (BLangArrowFunction) TreeBuilder.createArrowFunctionNode();
+        arrowFunction.pos = getPosition(implicitAnonymousFunctionExpressionNode);
+        arrowFunction.functionName = TreeBuilder.createIdentifierNode();
+        //TODO initialize other attributes
+        //arrowFunction.funcType;
+        //arrowFunction.function;
+
+        // Set Parameters
+        Node param = implicitAnonymousFunctionExpressionNode.params();
+        if (param.kind() == SyntaxKind.INFER_PARAM_LIST) {
+            try {
+                ImplicitAnonymousFunctionParameters paramsNode = (ImplicitAnonymousFunctionParameters) param;
+                SeparatedNodeList<SimpleNameReferenceNode> paramList = paramsNode.parameters();
+
+                for (SimpleNameReferenceNode child : paramList) {
+                    BLangUserDefinedType userDefinedType = (BLangUserDefinedType) child.apply(this);
+                    BLangSimpleVariable parameter = (BLangSimpleVariable) TreeBuilder.createSimpleVariableNode();
+                    parameter.name = userDefinedType.typeName;
+                    arrowFunction.params.add(parameter);
+                }
+            } catch (Exception e) {
```

Comment:
What is the `Exception` you are trying to catch? we should fix it properly form the parser side first.

Reply:
Removed try catch.